### PR TITLE
Update dataset.py

### PR DESCRIPTION
**Pull request by uakarsh**
For the last word, there is no relative distance feature, so i have added [0] to it

### CODE REVIEW

**Comment by uakarsh** [comment]
So, with this Pull request, we solve:
1. https://github.com/shabie/docformer/issues/19#issue-1087598707
2. https://github.com/shabie/docformer/issues/21

File: src/docformer/modeling.py [hunk 1]
```diff
@@ -54,44 +52,37 @@ def forward(self, x):
...
         self.x_topleft_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["coordinate_size"])
         self.x_bottomright_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["coordinate_size"])
         self.w_position_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_topleft_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_bottomleft_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_topright_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_bottomright_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
-        self.x_centroid_distance_to_prev_embeddings_v = nn.Embedding(config["max_2d_position_embeddings"], config["shape_size"])
+        self.x_topleft_distance_to_prev_embeddings_v = nn.Embedding(2*config["max_2d_position_embeddings"] + 1, config["shape_size"])
```
Comment (shabie) [inline src/docformer/modeling.py:72]: What's the reason for going for `2*config["max_2d_position_embeddings"] + 1` instead of the original?

Comment (uakarsh) [reply src/docformer/modeling.py:72]: The reason for this is mentioned below:
In the relative values, we are not taking the absolute values (following the conventions used by other algorithms such as LayoutLM, LayoutLMV2), instead what we do is:
1. Clamp the values ( let us say between -1024,1024) #1024 is the max_2d_position_embeddings 
2. Add a bias of 1024, so that now the values ranges between (0,2048)
3. Hence for the embedding layer, which are encoding the relative distance features, should havs a embedding layer range (0,2048, but I added 1, because of just remaining on the safe side

For the clamping, we have two options:
1. Either do it in the Extract Features of modeling.py
2. Or do it while making the dataset class

Currently, I thought of doing it in Dataset class, but maybe we can change it and write it in modeling.py file

**Comment by uakarsh** [comment]
Can you see this:  [Position Embedding in LayoutLM](https://github.com/huggingface/transformers/blob/master/src/transformers/models/layoutlm/modeling_layoutlm.py#L104)

I think that, for the position Encoding, this would seem fine (since they are learnable), rather than fix sin and cos function

The changes would be:

    `class PositionalEncoding(nn.Module):
            def __init__(self, d_model: int, dropout: float = 0.1, max_len: int = 5000):
                super().__init__()
                self.dropout = nn.Dropout(p=dropout)
                self.max_len = max_len
                self.position_embeddings = nn.Embedding(max_len, d_model)
                self.register_buffer("position_ids", torch.arange(max_len).expand((1, -1)))
        

        def forward(self) -> Tensor:
            x = self.position_ids[0, : self.max_len]
            x = self.position_embeddings(x)
            return self.dropout(x).unsqueeze(0)`
